bugsnag_test.go

```diff
@@ -23,15 +23,20 @@ type exception struct {
 	Message    string       `json:"message"`
 	Stacktrace []stackFrame `json:"stacktrace"`
 }
+
+type event struct {
+	Exceptions []exception      `json:"exceptions"`
+	Metadata   bugsnag.MetaData `json:"metaData"`
+}
+
 type notice struct {
-	Events []struct {
-		Exceptions []exception `json:"exceptions"`
-	} `json:"events"`
+	Events []event `json:"events"`
 }
 
 func TestNoticeReceived(t *testing.T) {
```

Comment:
I'd remove the code that tests the case where the metadata map is empty and add the checks for the values.